apps/fxc-front/src/app/workers/score-track.ts

```diff
@@ -0,0 +1,75 @@
+import {BRecord, IGCFile} from 'igc-parser';
+import {ScoreInfo, Scoring, scoringRules, solver} from 'igc-xc-score';
+import {RuntimeTrack} from "@flyxc/common";
+
+export interface Request {
+	track: RuntimeTrack;
+	league?: string;
+	config?: { [key: string]: any } | undefined;
+}
+
+export interface Response {
+	scoreInfo: ScoreInfo;
+	scoring: Scoring;
+	trackId: string;
+}
+
+const w: Worker = self as any;
+
+w.onmessage = (message: MessageEvent<Request>) => {
+	console.info('scoring: received', message);
+	try {
+		const request = message.data;
+		const flight = igcFile(request.track);
+		const solutions = solver(flight, getScoringRules(request), request.config);
+		let solution = solutions.next();
+		if (solution.value.scoreInfo){
+			let response: Response = {
+				scoreInfo: solution.value.scoreInfo,
+				scoring: {name:solution.value.opt.scoring.name, code: solution.value.opt.scoring.code, multiplier: solution.value.opt.scoring.multiplier},
+				trackId: message.data.track.id
+			}
+			console.info("scoring: computed", response.scoreInfo.score, response)
+			w.postMessage(response, {});
+		}
+	} catch (e) {
+		console.error('solver failed', e);
+	}
+};
+
+function getScoringRules(request:Request) {
+	let scoringRule;
+	switch (request.league) {
+		case "France (CFD)":
+			scoringRule = scoringRules.FFVL;
+			break;
+		case "XContest":
+			scoringRule = scoringRules.XContest;
+			break;
+		// TODO: others...
+		default :
+			scoringRule =  scoringRules.FFVL;
+	}
+	return scoringRule
+}
+
+
+// build a fake igc file from a track
+function igcFile(track: RuntimeTrack): IGCFile {
```

Comment:
rename to `createIgcFile()`

`fixes` should be `const` as not modified (lint should tell you)

The `Date` constructor expect a time in ms.

Doesn't `igc-xc-score` has an API that takes points instead of an `IGCFile`?

Why to we have `@ts-ignore`?

Reply:
fixed